_studio/mfx_lib/shared/src/libmfxsw_query.cpp

```diff
@@ -44,6 +45,13 @@ mfxStatus MFXQueryIMPL(mfxSession session, mfxIMPL *impl)
     {
         currentImpl = (mfxIMPL) (MFX_IMPL_HARDWARE2 + (session->m_adapterNum - 1));
     }
+
+    if (currentImpl == MFX_IMPL_HARDWARE &&
+        !CheckForIntelHWDisplayDevicePresent(session->m_adapterNum, session->m_implInterface))
```

Comment:
This is also needed on MFXInit. And I suggest to cache gathered information to avoid repeated operations.